### DISCUSSION TITLE
HA in Kubernetes

### DISCUSSION THREAD

**Discussion by R-Nabil** [TwiN/gatus#753] [General]
Hey

Is there any plan for gatus to become HA ?
It does feel like it would be a very useful feature as if the node hosting gatus itself in k3s becomes unavailable in a multi node setup then gatus stops monitoring anything until it is redeployed (which could take a bit of time)

Thanks

**Comment by Marck** [comment]
Havent tried this, but isnt it already possible if you set multiple replicas with selectors so it lands on different hosts?

**Reply by R-Nabil** [reply]
Well no. Unless you duplicate databases etc which defeats the purpose entirely 

I'd think that ideally you want an HA database behind and an HA gatus with a way to split the "tasks" between available replicas to avoid duplication of ping/status checks ?
